src/crystal/system/win32/file_descriptor.cr

```diff
@@ -11,7 +11,22 @@ module Crystal::System::FileDescriptor
   @system_blocking = true
 
   private def unbuffered_read(slice : Bytes)
-    if system_blocking?
+    handle = windows_handle
+    if LibC.GetConsoleMode(handle, nil) != 0
+      u16buffer = Slice(UInt16).new(slice.size)
+      chars_read = LibC::DWORD.new(0)
+      if 0 == LibC.ReadConsoleW(handle, u16buffer, u16buffer.size, pointerof(chars_read), nil)
+        raise IO::Error.from_winerror("Error reading from console")
+      end
+
+      appender = slice.to_unsafe.appender
+      String.each_utf16_char(u16buffer[0, chars_read]) do |char|
+        char.each_byte do |byte|
+          appender << byte
+        end
+      end
```

Comment:
I think this can touch OOB memory if the read string's UTF-8 representation is longer than its UTF-16 representation